src/cl/common.cl

```diff
@@ -28,11 +28,11 @@ uint reverse_bytes(uint a) {
          (a & 0x00ff0000) >> 8 | (a & 0xff000000) >> 24;
 }
 
-sha256_block hash_prefix(uint layer_index, uint node_index, uint window_index, replica_id id) {
+sha256_block hash_prefix(uint layer_index, ulong node_absolute_index, replica_id id) {
   sha256_block data = sha256_ZERO;
   data.vals[0] = layer_index;
-  data.vals[1] = node_index;
-  data.vals[2] = window_index;
+  data.vals[1] = node_absolute_index >> 32;
+  data.vals[2] = node_absolute_index;
```

Comment:
It's my belief that this is 'okay' because OpenCL guarantees consistent endianness of arguments once they are received by kernels. Assuming that is correct, can you please add a comment here making clear why we can safely use bit-shifting on integers to arrange the hash prefix?